File: experiments_dataset_vul_detect_cwe_classification/dataset/vul_dataset/java/cwe-20/code_before3.java
// Source: Row 29 in ./dataset/CVEfixes/Analysis/results/Java/df_java_cwe_20.xlsx

private void add0(int h, int i, final CharSequence name, final CharSequence value) {
if (!(name instanceof AsciiString)) {
    HttpUtils.validateHeader(name);
}
if (!(value instanceof AsciiString)) {
    HttpUtils.validateHeader(value);
}

File: experiments_dataset_vul_detect_cwe_classification/dataset/vul_dataset/javascript/cwe-20/code_before3.js
// Source: Row 55 in ./dataset/CVEfixes/Analysis/results/JavaScript/df_javascript_cwe_20.xlsx

URI.parseUserinfo = function(string, parts) {
    // extract username:password
    var firstBackSlash = string.indexOf('\\');
    var firstSlash = string.indexOf('/');
    var slash = firstBackSlash === -1 ? firstSlash : (firstSlash !== -1 ? Math.min(firstBackSlash, firstSlash): firstSlash)
    var pos = string.lastIndexOf('@', firstSlash > -1 ? firstSlash : string.length - 1);
    var t;

    // authority@ must come before /path or \path
    if (pos > -1 && (slash === -1 || pos < slash)) {
      t = string.substring(0, pos).split(':');
      parts.username = t[0] ? URI.decode(t[0]) : null;
      t.shift();
      parts.password = t[0] ? URI.decode(t.join(':')) : null;
      string = string.substring(pos + 1);
    } else {
      parts.username = null;
      parts.password = null;
    }

    return string;
  };